core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java

```diff
@@ -146,4 +148,58 @@ public PrimitiveQueryType(String name, String parentname, int blockIndex,
     return actualData;
   }
 
+  @Override public Object getDataBasedOnColumn(ByteBuffer dataBuffer, CarbonDimension parent,
+      CarbonDimension child) {
+    Object actualData = null;
+
+    if (parent.getOrdinal() != child.getOrdinal() || null == dataBuffer) {
+      return null;
+    }
+
+    if (isDirectDictionary) {
```

Comment:
Remove the duplicate code, code is almost similar to `getDataBasedOnDataType`

Reply:
removed and changed